packages/preview2-shim/lib/io/worker-http.js

```diff
@@ -127,7 +124,7 @@ export async function createHttpRequest(
           host: authority.split(":")[0],
           port: authority.split(":")[1],
           path: pathWithQuery,
-          timeout: connectTimeout && Number(connectTimeout),
+          timeout: connectTimeout && Number(connectTimeout / 1_000_000n),
```

Comment:
I think the `firstByteTimeout` and `betweenBytesTimeout` need this division step as well?

https://github.com/bytecodealliance/jco/blob/961feb73c3208cf1933db3e6b75ccb6f641b85ae/packages/preview2-shim/lib/io/worker-http.js#L150-L154

Reply:
Great catch, thanks, added https://github.com/bytecodealliance/jco/pull/409/.